crypto/asn1/asn1_lib.c

```diff
@@ -389,3 +389,35 @@ unsigned char *ASN1_STRING_data(ASN1_STRING *x)
     return x->data;
 }
 #endif
+
+char *sk_ASN1_UTF8STRING2text(STACK_OF(ASN1_UTF8STRING) *text, char sep,
+                              int max_len)
+{
+    int i;
+    ASN1_UTF8STRING *current;
+    int length = 0;
+    char *result = NULL;
+    char *p;
+
+    for (i = 0; i < sk_ASN1_UTF8STRING_num(text); ++i) {
+        current = sk_ASN1_UTF8STRING_value(text, i);
+        if (ASN1_STRING_length(current) > max_len - length - 1)
+            return NULL;
+        length += ASN1_STRING_length(current);
+        length += 1; /* separator character */
+    }
+    if ((result = OPENSSL_malloc(length)) == NULL)
```

Comment:
Looks like there is an off-by-one error if the text stack has no elements in it, i.e. `length` is 0, which does not allow enough space for the NUL terminator.

Reply:
Indeed there was an issue with empty input stack. When moving the existing code to here I did not really scan the code for potential errors.

I've fixed it and on this occasion generalized the `sep` parameter to be a string (rather than a single character).

Reply:
On this occasion I fixed a dozen of coding style nits in `asn1_lib.c`,
making use of the now official tool `check-format.pl`.